qt-models/divepicturemodel.cpp

```diff
@@ -56,17 +56,29 @@ void DivePictureModel::updateDivePictures()
 		endRemoveRows();
 	}
 
-	// if the dive_table is empty, ignore the displayed_dive
-	if (dive_table.nr == 0 || dive_get_picture_count(&displayed_dive) == 0)
+	// if the dive_table is empty, quit
+	if (dive_table.nr == 0)
 		return;
 
-	FOR_EACH_PICTURE_NON_PTR(displayed_dive)
-		pictures.push_back({picture, picture->filename, QImage(), picture->offset.seconds});
+	int i, pictures_count_old;
+	struct dive *dive;
+	for_each_dive (i, dive) {
+		if (dive->selected) {
+			pictures_count_old = pictures.count() - 1;
+			FOR_EACH_PICTURE(dive)
+				pictures.push_back({picture, picture->filename, QImage(), picture->offset.seconds});
 
-	QtConcurrent::blockingMap(pictures, scaleImages);
+			if (dive->id == displayed_dive.id) {
+				rowDDStart = pictures_count_old;
+				rowDDEnd = pictures.count() - 1;
+			}
 
-	beginInsertRows(QModelIndex(), 0, pictures.count() - 1);
-	endInsertRows();
+			QtConcurrent::blockingMap(pictures, scaleImages);
+
+			beginInsertRows(QModelIndex(), pictures_count_old, pictures.count() - 1);
```

Comment:
These two lines should also be moved out of the for_each_dive loop.
`pictures_count_old` is not necessary as far as I can tell. The idea of `beginInsertRows()` and `endInsertRows()` is to call them only once per batch-manipulation.